scripts/snippets/inf-hosp.r

```diff
@@ -0,0 +1,19 @@
+## first, choose random delays
+inf_hosp <- infection_times |>
+  mutate(
+    incubation_period = rgamma(n(), shape = 5, rate = 1),
+    onset_to_hosp = rlnorm(n(), meanlog = 1, sdlog = 1),
+    infection_date = as.Date(start_date + infection_time),
+    onset_date = as.Date(start_date + infection_time + incubation_period),
+    hosp_date =
+      as.Date(start_date + infection_time + incubation_period + onset_to_hosp)
```

Comment:
Is it going to be clear that `as.Date` acts as a flooring mechanism?

Reply:
Possibly not. I'm not a big fan of this either. Do you have a suggestion for a clearer way of doing this (alternatively can add an explanation).

Reply:
Turns out it actually doesn't, and this was news to me. We may have to use `lubridate` or one way or the other definitely something clearer.

``` r
date <- as.Date("2024-05-24")
fractional <- as.Date(date + 0.2)
fractional
#> [1] "2024-05-24"
as.POSIXct(fractional)
#> [1] "2024-05-24 04:48:00 UTC"
```

<sup>Created on 2024-05-24 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>